### PR TITLE
R&D machinery improvements

### PR DESCRIPTION
**Pull request by LagannPM**
Autolathe, protolathe/circuit imprinter, mech fab now has icons next to items
Pings on queue completion
Various fixes and improvements
default
default
default

### CODE REVIEW

File: cev_eris.dme [hunk 1]
```diff
@@ -298,6 +297,7 @@
 #include "code\datums\music_tracks\marhaba.dm"
 #include "code\datums\music_tracks\one_loop.dm"
 #include "code\datums\music_tracks\space_oddity.dm"
+#include "code\datums\objective\abduct.dm"
```
Comment (NanakoAC) [inline cev_eris.dme]: Why the fuck is this happening constantly
it happens on my PRs too, it always thinks ive moved this file

File: code/_helpers/text.dm [hunk 2]
```diff
@@ -50,6 +50,9 @@
 
 	return input
 
+/proc/sanitizeFileName(var/input)
```
Comment (NanakoAC) [inline code/_helpers/text.dm]: Did you really need to create this? Surely sanitising functions already exist?

Comment (LagannPM) [reply code/_helpers/text.dm]: Yes, its for file names, there is no analog

File: cev_eris.dme (hunk 1 above)
Comment (LagannPM) [reply cev_eris.dme]: have no clue

**Pull request merged**